python/lsst/meas/base/afterburner.py

```diff
@@ -0,0 +1,183 @@
+
+class AbContext(object):
+    '''
+    Context manager to handle catching errors that may have been thrown in an afterburner plugin
+    @param[in] plugin   The plugin that is to be run
+    @param[in] cat      Either a catalog or a source record entry of a catalog, depending of the plugin type,
+                        i.e. either working on a whole catalog, or a single record. 
+    @param[in] log      The log which to write to, most likely will always be the log (self.log) of the object
+                        in which the context manager is used.
+    '''
+    def __init__(self, plugin, cat, log):
+        self.plugin = plugin
+        self.cat = cat
+        self.log = log
+
+    def __enter__(self):
+        return
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        if exc_type is None:
+            return True
+        if exc_type in FATAL_EXCEPTIONS:
+            raise exc_value
+        elif exc_type is MeasurementError:
+            self.plugin.fail(self.cat, exc_value)
+        else:
+            self.log.warn("Error in {}.burn: {}".format(self.plugin.name, exc_value))
+        return True
+
+class AfterburnerConfig(lsst.pex.config.Config):
+    '''
+    Default AfterburnerConfig. Currently this is an empty list, meaning that there are no default plugins run.
+    The config object for each plugin must use this variable to specify the names of all plugins to be run.
+    '''
+    plugins = AfterburnerPlugin.registry.makeField(multi=True, default=[], doc="Plugins to be run and their "
+                                                    "configuration")
+    pass
+
+class AfterburnerTask(lsst.pipe.base.Task):
+    '''
+    This task facilitates running plugins which will operate on a source catalog. These plugins may do things
+    such as classifying an object based on source record entries inserted during a measurement task.
+
+    Plugins may either take an entire catalog to work on at a time, or 
+    '''
+    ConfigClass = AfterburnerConfig
+    _DefaultName = "afterburner"
+
+    def __init__(self, schema, plugMetadata=None, **kwargs):
+        """
+        Constructor; only called by derived classes
+
+        @param[in] plugMetaData     An lsst.daf.base.PropertyList that will be filled with metadata
+                                    about the plugins being run. If None, an empty empty PropertyList
+                                    will be created.
+        @param[in] **kwargs         Additional arguments passed to lsst.pipe.base.Task.__init__.
+        """
+        lsst.pipe.base.Task.__init__(self,**kwargs)
+        self.schema = schema
+        if plugMetadata is None:
+            plugMetadata = lsst.daf.base.PropertyList()
+        self.plugMetadata = plugMetadata
+
+        self.initializePlugins()
+
+    def initializePlugins(self):
+        '''
+        Initialize the plugins according to the configuration.
+        '''
+
+        pluginType = namedtuple('pluginType','single multi')
```

Comment:
I think `Struct` is better than `namedtuple`.  There's a bit more setup, but it's a traditional class so you can look at the code and add methods if appropriate.  If you do go with `namedtuple`, note that because it's a class the variable name (and the first argument) should be Capitalised.

    class PluginType(Struct):
        def __init__(self):
            Struct.__init__(self, single=[], multi=[])